merkletree_test.go

```diff
@@ -0,0 +1,312 @@
+package merkletree
+
+import (
+	"bytes"
+	"crypto/rand"
+	"reflect"
+	"testing"
+
+	"github.com/coniks-sys/libmerkleprefixtree-go/internal"
+
+	"golang.org/x/crypto/ed25519"
+	"golang.org/x/crypto/sha3"
+)
+
+var treeNonce = []byte("TREE NONCE")
+var salt = []byte("salt")
+
+var (
+	pk, sk []byte
+)
+
+func init() {
+	pk, sk, _ = ed25519.GenerateKey(rand.Reader)
+}
+
+func TestOneEntry(t *testing.T) {
+	m := InitMerkleTree(&DefaultPolicies{}, treeNonce, salt, pk, sk)
+
+	history := NewHistory(m, 1, 1)
```

Comment:
It'd be nice to be able to test MerkleTree on its own in merkletree_test.go
and then have separate tests for History in history_test.go